plugins/woocommerce/src/StoreApi/Utilities/JsonWebToken.php

```diff
@@ -50,6 +50,30 @@ public static function create( array $payload, string $secret ) {
 	 * @return bool
 	 */
 	public static function validate( string $token, string $secret ) {
+		if ( ! self::shallow_validate( $token ) ) {
+			return false;
+		}
+
+		$parts = self::get_parts( $token );
+
+		/**
+		 * Check if the token is based on our secret.
+		 */
+		$encoded_regenerated_signature = self::to_base_64_url(
+			self::generate_signature( $parts->header_encoded . '.' . $parts->payload_encoded, $secret )
+		);
+
+		return hash_equals( $encoded_regenerated_signature, $parts->signature_encoded );
+	}
+
+	/**
+	 * Shallow validate a token, it does not check the signature or expiration, but it checks the structure and expiry.
+	 *
+	 * @param string $token Full token string.
+	 *
+	 * @return bool
+	 */
+	public static function shallow_validate( string $token ) {
```

Comment:
I needed a way to validate that a token is structurally valid without checking the secret part, which is fine.